integreat_cms/cms/utils/repair_tree.py

```diff
@@ -0,0 +1,284 @@
+"""
+This module contains utilities to repair or detect inconsistencies in a tree
+"""
+
+from __future__ import annotations
+
+import logging
+from collections.abc import Callable
+from typing import Generator
+
+from ..models import Page
+from ..utils.tree_mutex import tree_mutex
+
+logger = logging.getLogger(__name__)
+
+
+class Printer:
+    """
+    Select printer for stdout or log file
+    """
```

Comment:
Only because you jumped at it while we were still working on it :slightly_smiling_face: You're probably right, but I think we'll try to actually get it in working condition first